extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/HttpInstrumenterVertxTracer.java

```diff
@@ -0,0 +1,461 @@
+        public List<String> requestHeader(final HttpRequest request, final String name) {
+            return request.headers().getAll(name);
+        }
+
+        @Override
+        public Long requestContentLength(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public Long requestContentLengthUncompressed(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public Integer statusCode(final HttpRequest request, final HttpResponse response) {
+            return response.statusCode();
+        }
+
+        @Override
+        public Long responseContentLength(final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public Long responseContentLengthUncompressed(
+                final HttpRequest request, final HttpResponse response) {
+            return null;
+        }
+
+        @Override
+        public List<String> responseHeader(final HttpRequest request, final HttpResponse response, final String name) {
+            return response.headers().getAll(name);
+        }
+    }
+
+    private static class HttpRequestTextMapSetter implements TextMapSetter<HttpRequest> {
+        @Override
+        public void set(final HttpRequest carrier, final String key, final String value) {
+            if (carrier != null) {
+                carrier.headers().set(key, value);
+            }
+        }
+    }
+
+    static class HttpRequestSpan implements HttpRequest {
+        private final HttpRequest httpRequest;
+        private final MultiMap headers;
+        private final Context context;
+        private final io.opentelemetry.context.Context spanContext;
+
+        HttpRequestSpan(
+                final HttpRequest httpRequest,
+                final MultiMap headers,
+                final Context context,
+                final io.opentelemetry.context.Context spanContext) {
+
+            this.httpRequest = httpRequest;
+            this.headers = headers;
+            this.context = context;
+            this.spanContext = spanContext;
+        }
+
+        @Override
+        public int id() {
+            return httpRequest.id();
+        }
+
+        @Override
+        public String uri() {
+            return httpRequest.uri();
+        }
+
+        @Override
+        public String absoluteURI() {
+            return httpRequest.absoluteURI();
+        }
+
+        @Override
+        public HttpMethod method() {
+            return httpRequest.method();
+        }
+
+        @Override
+        public MultiMap headers() {
+            return headers;
+        }
+
+        @Override
+        public SocketAddress remoteAddress() {
+            return httpRequest.remoteAddress();
+        }
+
+        public Context getContext() {
+            if (context == null) {
+                throw new IllegalStateException();
```

Comment:
Can you add a message? I'm sure we will have to debug it one day, would be better to identify it quickly. 
Something like: "No context set".